src/app/google.service.ts

```diff
@@ -0,0 +1,86 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { AuthConfig, OAuthService } from 'angular-oauth2-oidc';
+import { Observable, Subject } from 'rxjs';
+
+const authCodeFlowConfig: AuthConfig = {
+  // Url of the Identity Provider
+  issuer: 'https://accounts.google.com',
+
+  // strict discovery document disallows urls which not start with issuers url
+  strictDiscoveryDocumentValidation: false,
+
+  // URL of the SPA to redirect the user to after login
+  redirectUri: window.location.origin,
+
+  // The SPA's id. The SPA is registerd with this id at the auth-server
+  // clientId: 'server.code',
+  clientId: '413745185671-d7d6gfcc07efqst5ft7cfsms41vqs28i.apps.googleusercontent.com',
+
+  // set the scope for the permissions the client should request
+  scope: 'openid profile email https://www.googleapis.com/auth/gmail.readonly',
+
+  showDebugInformation: true,
+};
+
+export interface UserInfo {
```

Comment:
Umieszczaj sobie interfejsy te które eksportujesz w osobnych plikach - ułatwia to potem refaktor kodu.